integrationTests/DonationRepositoryTest.php

```diff
@@ -234,7 +234,15 @@ public function testPushSalesForcePendingPushesNewDonationButNotVeryNewDonation(
 
         // assert
         $busDispatchMethod = $busProphecy->dispatch(Argument::type(Envelope::class))
-            ->willReturn($dummyEnvelope);
+            ->will(function (array $args) use ($donationUUID) {
+                /** @var Envelope $envelope */
+                $envelope = $args[0];
+                $message = $envelope->getMessage();
+                \assert($message instanceof DonationUpserted);
+                \assert($message->uuid === $donationUUID);
```

Comment:
All very minor points but:
- probably better to put the type on the param and let the local be inferred from that instead of typing the local:
- and for assertions that are the point of the test (rather than assertions that are just needed incidentally) a bit better to use the PHPUnit assertions instead of built in `assert`. I realise we can't use $this in these closures because its rebound, but these are static functions anyway so we don't need to.

```suggestion
            ->will(/**
             * @param array{0: Envelope} $args
             */ function (array $args) use ($donationUUID) {
                $envelope = $args[0];
                $message = $envelope->getMessage();
                TestCase::assertInstanceOf(DonationUpserted::class, $message);
                TestCase::assertSame($donationUUID, $message->uuid);
```